### ISSUE TITLE
Audit files in XPI

### ISSUE THREAD

**Issue by pdehaan** [mozilla/blok#69]
Unzipping the **/web-ext-artifacts/blok-0.1.7-fx.xpi**, there are a few interesting results:

```sh
$ unzip blok-0.1.7-fx.xpi
$ tree

├── META-INF/
│   ├── manifest.mf
│   ├── mozilla.rsa
│   └── mozilla.sf
├── README.md
├── coverage/ (EMPTY)
├── css/
│   ├── contentscript.css
│   ├── feedback.css
│   ├── foundation.min.css
│   └── toolbar.css
├── disconnect-blocklist.json
├── disconnect-entitylist.json
├── feedback.html
├── icons/ (EMPTY)
├── img /
│   ├── banner.png
│   ├── blok.png
│   ├── screenshot.png
│   ├── tracking-protection-16.svg
│   └── tracking-protection-disabled-16.svg
├── js/
│   ├── background.bundle.js
│   ├── background.js
│   ├── canonicalize.js
│   ├── contentscript.js
│   ├── feedback.js
│   ├── lists.js
│   ├── log.js
│   └── toolbar.js
├── manifest.json
├── package.json
├── tests/
│   ├── blocklist-fixture.json
│   ├── test-canonicalize.js
│   └── test-lists.js
├── toolbar.html
└── web-ext-artifacts/ (EMPTY)

8 directories, 30 files
```

```sh
$ ls -lh

drwxr-xr-x   5 pdehaan  staff   170B Jul 28 16:28 META-INF
-rw-r--r--   1 pdehaan  staff   1.6K Jul 27 14:48 README.md
drwxr-xr-x   2 pdehaan  staff    68B Jul 27 14:48 coverage/
drwxr-xr-x   6 pdehaan  staff   204B Jul 27 14:48 css/
-rw-r--r--   1 pdehaan  staff   260K Jul 27 14:48 disconnect-blocklist.json
-rw-r--r--   1 pdehaan  staff   196K Jul 27 14:48 disconnect-entitylist.json
-rw-r--r--   1 pdehaan  staff   1.7K Jul 27 14:48 feedback.html
drwxr-xr-x   2 pdehaan  staff    68B Jul 27 14:48 icons/
drwxr-xr-x   7 pdehaan  staff   238B Jul 27 14:48 img/
drwxr-xr-x  10 pdehaan  staff   340B Jul 27 14:48 js/
-rw-r--r--   1 pdehaan  staff   719B Jul 27 14:48 manifest.json
-rw-r--r--   1 pdehaan  staff   1.6K Jul 27 14:48 package.json
drwxr-xr-x   5 pdehaan  staff   170B Jul 27 14:48 tests/
-rw-r--r--   1 pdehaan  staff   1.7K Jul 27 14:48 toolbar.html
drwxr-xr-x   2 pdehaan  staff    68B Jul 27 14:48 web-ext-artifacts/
```

Most curious are the empty `/coverage/`, `/icons/` directories (/icons/ was renamed to /img/ in #63), and `/web-ext-artifacts/`.

I'm guessing now that we've merged #43 we'll also have a /docs/ directory in the next build.

We _probably_ (maybe?) can remove the following things from the generated XPI add-on:
- /coverage/
- /docs/*
- /icons/
- /tests/*
- /web-ext-artifacts/
- /img/*.png (add-on may only need *.svg files)
- /js/background.js (**?** since we have a browserified /js/background.bundle.js)
- /README.md
- /manifest.json (**?** not sure if needed in the .XPI or just to build the XPI)
- /package.json (**?** not sure if needed in the .XPI or just to build the XPI)

**Comment by groovecoder** [comment]
Is this something we can do (easily) in/with `manifest.json` directives for `web-ext`?

Adding config/build complexity to save __ kilobytes from the `.xpi` feels like a low priority.

**Comment by pdehaan** [comment]
Actually, this currently may be blocked by lack of file filtering in **web-ext** (a la .jpmignore in **jpm**). I think we can safely ignore this until I can figure out status on some upstream bugs.

❤️

**Comment by pdehaan** [comment]
> Adding config/build complexity to save __ kilobytes from the .xpi feels like a low priority.

Actually the ominous `__` in your comment made me curious, so here's the report of what the current file sizes would theoretically be:

```sh
$ ls -lash web-ext-artifacts/blok-0.1.7*
416 -rw-r--r--  1 pdehaan  staff   205K Jul 29 11:14 web-ext-artifacts/blok-0.1.7-dev.zip
168 -rw-r--r--  1 pdehaan  staff    84K Jul 29 11:27 web-ext-artifacts/blok-0.1.7-edited.zip
424 -rw-r--r--  1 pdehaan  staff   211K Jul 29 10:46 web-ext-artifacts/blok-0.1.7-fx.xpi
416 -rw-r--r--  1 pdehaan  staff   205K Jul 29 11:24 web-ext-artifacts/blok-0.1.7-prod.zip
```

Where:

| FILE | SIZE | COMMENTS |
|:----|----:|:----|
| blok-0.1.7-fx.xpi | 211KB | Current built in the repo. Presumably signed by AMO.
| blok-0.1.7-prod.zip | 205KB | Built using `npm run build`.
| blok-0.1.7-dev.zip | 205KB | Built using `npm run build-dev`. Oddly same size as the `npm run build` ZIP.
| blok-0.1.7-edited.zip | 84KB | Removed unused/empty files, and PNGs, etc.

![blok-0_1_7-edited_zip](https://cloud.githubusercontent.com/assets/557895/17259443/5c9319b6-5580-11e6-910d-7348a8a20fe9.png)

**Comment by pdehaan** [comment]
From IRC w/ kumar:

- **pdehaan>** can the latest web-ext ignore arbitrary files/folder via the `web-ext build` command?

- **kumar>** not currently. It's unfortunately not a trivial patch https://github.com/mozilla/web-ext/issues/131

- **kumar>** if you are using web-ext as a library then it should be pretty easy. You can pass in custom minimatch rules https://github.com/mozilla/web-ext/blob/master/src/cmd/build.js#L144
